### PR TITLE
OSDOCS#10121: Updating for removed Kubernetes APIs in 1.29

### PR DESCRIPTION
**Pull request by bergerhoffer**
Version(s):
4.16

Issue:
https://issues.redhat.com/browse/OSDOCS-10121

Link to docs preview:
* https://74097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/updating-cluster-prepare.html#kube-api-removals_updating-cluster-prepare
* https://74097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/updating_a_cluster/updating-cluster-cli#prerequisites

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/update-preparing-evaluate-apirequestcount-workloads.adoc [hunk 1]
```diff
@@ -41,7 +41,5 @@ $ oc get apirequestcounts csistoragecapacities.v1beta1.storage.k8s.io \
 [source,terminal]
 ----
 VERBS       USERNAME                        USERAGENT
-list watch  system:kube-controller-manager  cluster-policy-controller/v0.0.0
-list watch  system:kube-controller-manager  kube-controller-manager/v1.26.5+0abcdef
-list watch  system:kube-scheduler           kube-scheduler/v1.26.5+0abcdef
+// TODO: Need output, no hits in test cluster
```
Comment (bergerhoffer) [inline modules/update-preparing-evaluate-apirequestcount-workloads.adoc:44]: @wangke19 @sanchezl Do you know if any good example we can show here? My test cluster didn't have any hits, so I wasn't getting any output from this command.

Comment (bergerhoffer) [reply modules/update-preparing-evaluate-apirequestcount-workloads.adoc:44]: @sanchezl Any chance you could help with some example output here?

Comment (sanchezl) [reply modules/update-preparing-evaluate-apirequestcount-workloads.adoc:44]:
```suggestion
VERBS     USERNAME                            USERAGENT
create    andrea                              Mozilla/5.0 (X11; Linux x86_64)
create    system:admin                        oc/4.13.0 (linux/amd64)
list get  system:serviceaccount:myns:default  oc/4.17.0 (linux/amd64)
watch     system:serviceaccount:myns:webhook  webhook/v1.0.0 (linux/amd64)
```

**Comment by bergerhoffer** [comment]
@wangke19 Can you please QE review?

Preview: https://74097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/updating-cluster-prepare.html#kube-api-removals_updating-cluster-prepare

**Review by mletalie — COMMENTED** [review]
/label peer-review-in-progress

File: updating/preparing_for_updates/updating-cluster-prepare.adoc [hunk 2]
```diff
@@ -27,15 +27,11 @@ Without the correct micro-architecture requirements, the update process will fai
...
+A cluster administrator must provide a manual acknowledgment before the cluster can be updated from {product-title} 4.15 to 4.16. This is to help prevent issues after upgrading to {product-title} 4.16, where APIs that have been removed are still in use by workloads, tools, or other components running on or interacting with the cluster. Administrators must evaluate their cluster for any APIs in use that will be removed and migrate the affected components to use the appropriate new API version. After this evaluation and migration is complete, the administrator can provide the acknowledgment.
 
-Before you can update your {product-title} 4.13 cluster to 4.14, you must provide the administrator acknowledgment.
+Before you can update your {product-title} 4.15 cluster to 4.16, you must provide the administrator acknowledgment.
```
Comment (mletalie) [inline updating/preparing_for_updates/updating-cluster-prepare.adoc:34]: Not sure if this sentence needs to be here given the paragraph above?

**Review by mletalie — COMMENTED** [review]
/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done
/lgtm

**Comment by mletalie** [comment]
Hello @bergerhoffer,
   Hope all is well. Just one comment, otherwise LGTM. Great job!

File: updating/preparing_for_updates/updating-cluster-prepare.adoc (hunk 2 above)
Comment (bergerhoffer) [reply updating/preparing_for_updates/updating-cluster-prepare.adoc:34]: Good point, it is a bit repetitive. We have had this in all past releases like this... I'll think on whether to remove it or just leave it as is for consistency. Thanks!

**Comment by bergerhoffer** [comment]
Sending for peer-review again because I added a few additional changes to some files.

Here is the diff if that's helpful: https://github.com/openshift/openshift-docs/compare/8f34227be6ea4ca9e0d244efbc024a24bfce8faf..3e885405f9bf80220ae9a5636a9d72671a705072

**Review by mletalie — COMMENTED** [review]
/label peer-review-done
/remove-label peer-review-in-progress
/remove-label peer-review-needed
/lgtm

**Comment by mletalie** [comment]
Hello @bergerhoffer, I think you may want to update your preview links in the  "Links to docs preview" section b/c of the additional changes you made, otherwise LGTM!

**Comment by wangke19** [comment]
> @wangke19 Can you please QE review?
> 
> Preview: https://74097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/updating-cluster-prepare.html#kube-api-removals_updating-cluster-prepare

One minor change in lines https://github.com/openshift/openshift-docs/pull/74097/files#diff-592fca9656842e93d754c6e8df0dff28ce702d6979e7b8c45ab6088c9fdfd047R53-R54, I checked with latest 4.16 nightly, the actual results as below,
```
$ oc get apirequestcounts -o jsonpath='{range .items[?(@.status.removedInRelease!="")]}{.status.removedInRelease}{"\t"}{.metadata.name}{"\n"}{end}'
1.26	flowschemas.v1beta1.flowcontrol.apiserver.k8s.io
1.26	prioritylevelconfigurations.v1beta1.flowcontrol.apiserver.k8s.io

```

**Comment by bergerhoffer** [comment]
> > @wangke19 Can you please QE review?
> > Preview: https://74097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/updating-cluster-prepare.html#kube-api-removals_updating-cluster-prepare
> 
> One minor change in lines https://github.com/openshift/openshift-docs/pull/74097/files#diff-592fca9656842e93d754c6e8df0dff28ce702d6979e7b8c45ab6088c9fdfd047R53-R54, I checked with latest 4.16 nightly, the actual results as below,
> 
> ```
> $ oc get apirequestcounts -o jsonpath='{range .items[?(@.status.removedInRelease!="")]}{.status.removedInRelease}{"\t"}{.metadata.name}{"\n"}{end}'
> 1.26	flowschemas.v1beta1.flowcontrol.apiserver.k8s.io
> 1.26	prioritylevelconfigurations.v1beta1.flowcontrol.apiserver.k8s.io
> ```

@wangke19 So the output you're showing here references ones that were removed in 1.26 (v1beta1), not 1.29 (v1beta2), so I didn't think that would be correct to show as an example. wdyt?

**Review by sanchezl — APPROVED** [review]
/lgtm

**Comment by wangke19** [comment]
> > > @wangke19 Can you please QE review?
> > > Preview: https://74097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/updating-cluster-prepare.html#kube-api-removals_updating-cluster-prepare
> > 
> > 
> > One minor change in lines https://github.com/openshift/openshift-docs/pull/74097/files#diff-592fca9656842e93d754c6e8df0dff28ce702d6979e7b8c45ab6088c9fdfd047R53-R54, I checked with latest 4.16 nightly, the actual results as below,
> > ```
> > $ oc get apirequestcounts -o jsonpath='{range .items[?(@.status.removedInRelease!="")]}{.status.removedInRelease}{"\t"}{.metadata.name}{"\n"}{end}'
> > 1.26	flowschemas.v1beta1.flowcontrol.apiserver.k8s.io
> > 1.26	prioritylevelconfigurations.v1beta1.flowcontrol.apiserver.k8s.io
> > ```
> 
> @wangke19 So the output you're showing here references ones that were removed in 1.26 (v1beta1), not 1.29 (v1beta2), so I didn't think that would be correct to show as an example. wdyt?

@bergerhoffer This should not be a problem, the version here depends on the version still used by the client. Please ignore.

**Pull request merged**